### PR TITLE
Add Automation Guidance to Upgrade Documentation

### LINKED ISSUE #745: Add guidance for updating automation scripts for use with Chocolatey CLI v2.0.0+
## What New Or Updated Would You Like To See?

Create documentation with guidance for updating automation scripts for use with Chocolatey 2.x+

## Why Is It Needed?

Changes to Chocolatey CLI in v2.0.0 and beyond will break existing automation because of breaking changes.

### PR DESCRIPTION
**Pull request by steviecoaster**
## Description Of Changes

Adds guidance to upgrade documentation for Chocolatey CLI v2.0.0

## Motivation and Context

With the breaking changes introduced, automation scripts will break in customer environments with the upgrade to Chocolatey CLI v2.0.0

## Testing

* [x] I have previewed these changes using the [Docker Container](https://github.com/chocolatey/docs/tree/master/.devcontainer) or another method before submitting this pull request.

## Change Types Made

* [ ] Minor documentation fix (typos etc.).
* [x] Major documentation change (refactoring, reformatting or adding documentation to existing page).
* [ ] New documentation page added.
* [ ] The change I have made should have a video added, and I have raised an issue for this.
    * Issue #

## Change Checklist

* [ ] Requires a change to menu structure (top or left hand side)/
* [ ] Menu structure has been updated

## Related Issue

Fixes #745

### CODE REVIEW

File: input/en-us/guides/upgrading-to-chocolatey-v2-v6.md [hunk 1]
```diff
@@ -338,6 +338,20 @@ There are many ways to manage Chocolatey packages using integrations such as [An
 
 1. Upgrade to the latest stable 2.0.0 and 6.0.0 Chocolatey product versions.
 
+## Updating Automation Scripts For Chocolatey V2.0.0 Migration
```
Comment (pauby) [inline input/en-us/guides/upgrading-to-chocolatey-v2-v6.md:341]: I think this should be added up in the section that talks about -lo and --local-only being removed. I think adding it down here is going to cause it to be missed.

File: input/en-us/guides/upgrading-to-chocolatey-v2-v6.md [hunk 2]
```diff
@@ -338,6 +338,20 @@ There are many ways to manage Chocolatey packages using integrations such as [An
...
+
+### Interrogating local packages
+
+With the Chocolatey CLI v2.0.0 release the `choco list` command now only reports on locally-installed Chocolatey packages. If you are using the `-lo` or `--local-only` arguments with your `choco list` commands, these should be removed. If you fail to remove this argument, a warning will be displayed, which could cause your scripts to fail. For interrogating locally-installed packages we recommend `choco list` or `choco list --limit-output`, if you are looking for a more parsable format (recommended).
+
+In PowerShell `choco list --limit-output | ConvertFrom-Csv -Delimiter '|' -Header PackageId,PackageVersion` will return a PowerShell object you can more easily interact with in your automations.
+
+### Querying Remote Sources
+
+With the Chocolatey CLI v2.0.0 release it is now required to use the `choco search` command to query remote package information. If you are currently using the `choco list` command for this, update your automations to use `choco search`. If you are using `--limit-output` or `-r` as part of your command, this will continue to behave as normal.
```
Comment (pauby) [inline input/en-us/guides/upgrading-to-chocolatey-v2-v6.md:353]:
```suggestion
With the Chocolatey CLI v2.0.0 release it is now required to use the `choco search` command to query remote package information. If you are currently using the `choco list` command for this, update your automations to use `choco search`.
```

File: input/en-us/guides/upgrading-to-chocolatey-v2-v6.md [hunk 3]
```diff
@@ -338,6 +338,20 @@ There are many ways to manage Chocolatey packages using integrations such as [An
...
 
+## Updating Automation Scripts For Chocolatey V2.0.0 Migration
+
+The breaking changes introduced with the release of Chocolatey CLI V2.0.0 and Chocolatey Licensed Extension 6.0.0 mean that any automations that you are currently using will need to be updated.
+
+### Interrogating local packages
+
+With the Chocolatey CLI v2.0.0 release the `choco list` command now only reports on locally-installed Chocolatey packages. If you are using the `-lo` or `--local-only` arguments with your `choco list` commands, these should be removed. If you fail to remove this argument, a warning will be displayed, which could cause your scripts to fail. For interrogating locally-installed packages we recommend `choco list` or `choco list --limit-output`, if you are looking for a more parsable format (recommended).
+
+In PowerShell `choco list --limit-output | ConvertFrom-Csv -Delimiter '|' -Header PackageId,PackageVersion` will return a PowerShell object you can more easily interact with in your automations.
```
Comment (pauby) [inline input/en-us/guides/upgrading-to-chocolatey-v2-v6.md:349]: This is really good to call out!

**Review by pauby — CHANGES_REQUESTED** [review]

**Comment by steviecoaster** [comment]
@pauby if you want to update things here and take it over, that's fine!

**Comment by pauby** [comment]
@steviecoaster How does this look? Does it cover what you were looking for?

I split the `list` and `search` into separate sections as I felt it was starting to get confusing.

**Comment by pauby** [comment]
Awesome! Please merge!

**Pull request merged**